### ISSUE TITLE
InflateException when adding Preference's by code

### ISSUE THREAD

**Issue by dmmh** [Prototik/HoloEverywhere#416]
```
04-22 19:17:45.904: E/ACRA(5931): xxx.xxxxxx.xxx fatal error : Binary XML file line #2: Error inflating class <unknown>
04-22 19:17:45.904: E/ACRA(5931): android.view.InflateException: Binary XML file line #2: Error inflating class <unknown>
04-22 19:17:45.904: E/ACRA(5931):   at android.view.LayoutInflater.createView(LayoutInflater.java:613)
04-22 19:17:45.904: E/ACRA(5931):   at org.holoeverywhere.LayoutInflater.tryCreateView(LayoutInflater.java:357)
04-22 19:17:45.904: E/ACRA(5931):   at org.holoeverywhere.LayoutInflater.onCreateView(LayoutInflater.java:306)
04-22 19:17:45.904: E/ACRA(5931):   at android.view.LayoutInflater.onCreateView(LayoutInflater.java:660)
04-22 19:17:45.904: E/ACRA(5931):   at android.view.LayoutInflater.createViewFromTag(LayoutInflater.java:685)
04-22 19:17:45.904: E/ACRA(5931):   at android.view.LayoutInflater.inflate(LayoutInflater.java:466)
04-22 19:17:45.904: E/ACRA(5931):   at org.holoeverywhere.LayoutInflater.inflate(LayoutInflater.java:285)
04-22 19:17:45.904: E/ACRA(5931):   at android.view.LayoutInflater.inflate(LayoutInflater.java:396)
04-22 19:17:45.904: E/ACRA(5931):   at org.holoeverywhere.LayoutInflater.inflate(LayoutInflater.java:270)
04-22 19:17:45.904: E/ACRA(5931):   at org.holoeverywhere.preference.Preference.onCreateView(Preference.java:540)
04-22 19:17:45.904: E/ACRA(5931):   at org.holoeverywhere.preference.Preference.getView(Preference.java:426)
04-22 19:17:45.904: E/ACRA(5931):   at org.holoeverywhere.preference.PreferenceGroupAdapter.getView(PreferenceGroupAdapter.java:162)
04-22 19:17:45.904: E/ACRA(5931):   at org.holoeverywhere.widget.ListView$ListAdapterWrapper.getView(ListView.java:341)
```

The wierd thing is this only happens when I don't use a Activity context, but a application context. And it works fine passing application Context to a new PreferenceScreen. I've been trying to fix it for months, but can't find the culprit.  
I do need application context for a multitude of reasons, even though it doesn't show in the example.
And I am definately not going to switch to xml resource files, I like to handle Preference's by code. Really don't get why everybody uses xml, but maybe that's just me :)

CODE USED (simplified example)

```
public class Settings extends PreferenceActivity {
    private static Context context = ApplicationSubclass.getApplicationContext_();

    @Override
    public void onResume() {
        super.onResume();

        Log.d(getClass().getSimpleName(), "onResume()");
        setPreferenceScreen(createTest());
    }// end method

    private PreferenceScreen createTest() {
        PreferenceScreen root = getPreferenceManager().createPreferenceScreen(  context);

        PreferenceScreen screen1 = getPreferenceManager().createPreferenceScreen(context); // this works
        screen1.setTitle("test");
        screen1.setKey("screen1");
        root.addPreference(screen1);

        PreferenceScreen screen2 = getPreferenceManager().createPreferenceScreen(this); // this works
        screen2.setTitle("test");
        screen2.setKey("screen2");
        root.addPreference(screen2);

        CheckBoxPreference test1 = new CheckBoxPreference(this); // this also
                                                                // works
        test1.setTitle("SELECT1");
        test1.setKey(Data.KEY1);
        root.addPreference(test1);

        CheckBoxPreference test2 = new CheckBoxPreference(context); // this causes the error...!!!
        test2.setTitle("SELECT2");
        test2.setKey(Data.KEY2);
        root.addPreference(test2);

        return root;
    }// end method
}// end class
```

**Comment by Prototik** [comment]
For using in application context for creating preferences you should... F*ck, why are you use application context?

So, well, you should apply Holo.Theme\* on application context. If it no will work - wrap context by PreferenceInit.context(Context) method, like this:

``` java
    private static Context context = PreferenceInit.context(ApplicationSubclass.getApplicationContext_());
```

**Comment by dmmh** [comment]
Thanks for your reply. I looked at your suggestion and it seems I don't understand what you mean. I have the theme applied on application context.

using this in every variation I tried:
<application
        android:name=".subclasses.ApplicationSubclass"
        android:allowBackup="true"
        android:allowClearUserData="true"
        android:hasCode="true"
        android:icon="@drawable/ic_launcher"
        android:label="@string/app_name"
        android:theme="@style/Holo.Theme" >

My application subclass extends org.holoeverywhere.app.Application. Also tried android.app.Application. Doesn't make any difference.

I actually don't need application context on the Preference's itself, but I do need to pass the application context because I need it for other stuff, like registering receivers, and sending notifications with NotificationManager. Since there isn't an alternative constructor, this isn't possible. And a non-default constructor probably wouldn't be recommended anyway. This isn't far fetched, people would be activating and de-activating receivers via CheckBoxPreference quite often I guess. 

I will explain further. I am using a FragmentStatePagerAdapter. It's constructor creates the fragments, which are PreferenceFragments. Everything referencing HoloEverywhere ofcourse. The fragments create the PreferenceScreen's in the same way as shown above. I merely simplified the example, because it also doesn't work by code in a simple PreferenceActivity, which is unexpected behaviour. 

Also, it deviates from the default Preference framework. This is strange, because it really shouldn't make any difference what context is being used. In most cases, application context is more usefull anyway. I do still feel this is an issue that could be resolved. It's really weird having to pass in 2 contexts, when normally one is sufficient. Also, there is no way to wrap context (or set) on any of HoloEverywhere's Preference classes. Am I missing something?

If you mean I should modify the library itself, I don't think this is recommended. Also, since I am using this library in my own library project, I wouldn't get the right context, since the ApplicationSubclass won't work in a library. You must mean something else. Also, I don't see the point in wrapping anything, because it would still crash, as it's still the same application context.
